### PR TITLE
Admin Router: IAM tests and mock refactoring

### PR DESCRIPTION
**Pull request by vespian**
## High Level Description

This PR:
* backports changes from https://github.com/mesosphere/dcos-enterprise/pull/660 to Open:
  * simplify IAM mock
  * add some missing DC/OS mock-files
  * `auth.lua` fixes
* cleans up the usage of `EndpointException`
* removes dead code from `LineBufferFilter` utility
* refactors generic test code into separate common, open and ee parts

## Related Issues

  - [DCOS-14581](https://jira.dcos.io/browse/DCOS-14581) Admin Router: Authn+Authz tests for all the endpoints in Open
  - [DCOS-14582](https://jira.dcos.io/browse/DCOS-14582) Admin Router: Backport to Open IAM mock refactoring from EE

## Dependencies

EE DC/OS PR: https://github.com/mesosphere/dcos-enterprise/pull/757

### CODE REVIEW

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

File: packages/adminrouter/extra/src/test-harness/modules/util.py [hunk 1]
```diff
@@ -394,3 +375,52 @@ def ar_listen_link_setup(role, is_ee):
         return
 
     os.symlink(dst_path, src_path)
+
+
+@contextmanager
+def iam_denies_all_requests(mocker_instance):
```
Comment (mhrabovcin) [inline packages/adminrouter/extra/src/test-harness/modules/util.py]: 👍

File: packages/adminrouter/extra/src/test-harness/tests/test_auth.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+
+class TestAuthnJWTValidator:
+    """Tests scenarios where authentication token isn't provided or is provided
+    in different supported places (cookie, header)"""
+
+    def test_auth_token_not_provided(self, master_ar_process):
+        log_messages = {
+            # "Authorization header not found.": SearchCriteria(1, True),
+            # "dcos-acs-auth-cookie not found.": SearchCriteria(1, True),
```
Comment (mhrabovcin) [inline packages/adminrouter/extra/src/test-harness/tests/test_auth.py]: I've removed these commented out log messages from `ee` version.

Comment (vespian) [inline packages/adminrouter/extra/src/test-harness/tests/test_auth.py]: We should probably remove them from Open to so that they do not bit-rot.

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

**Comment by vespian** [comment]
This one is ready to go!

**Comment by orsenthil** [comment]
it is not required now. For future it is better to keep commits squashed to single one when possible. (Helps with history tracking).

**Pull request merged**